File: AuthHub.Models/Entities/Passwords/ClaimsKey.cs
using Common.Models.Entities;
using System;

namespace AuthHub.Models.Entities.Passwords
{
    public class ClaimsKey : EntityBase<int>
    {
        public string Name { get; set; }
        public string DefaultValue { get; set; }
        public int AuthSettingsId { get; set; }
        public bool IsDefault { get; set; }

        public ClaimsKey()
        {
        }

        public ClaimsKey(
            string name,
            int authSettingsId,
            int id
            ) : this()
        {
            Name = name;
            AuthSettingsId = authSettingsId;
            Id = id;
        }
    }
}

File: authhub-ui/src/app/models/ClaimsKey.ts
export interface ClaimsKey {
    id: number; // Assuming EntityBase<int> translates to an `id` property
    name: string;
    defaultValue: string;
    authSettingsId: number;
    isDefault: boolean;
}

// Example of a constructor if you prefer using a class
export class ClaimsKeyModel implements ClaimsKey {
    constructor(
        public id: number,
        public name: string,
        public defaultValue: string,
        public authSettingsId: number,
        public isDefault: boolean
    ) {}
}

// Usage example
const exampleClaimsKey: ClaimsKey = {
    id: 1,
    name: 'Example Claim',
    defaultValue: 'Default Value',
    authSettingsId: 1,
    isDefault: true
};
